visualization/src/point_cloud_handlers.cpp

```diff
@@ -686,56 +686,53 @@ pcl::visualization::PointCloudGeometryHandler<pcl::PCLPointCloud2>::getGeometry
 
   vtkSmartPointer<vtkFloatArray> data = vtkSmartPointer<vtkFloatArray>::New ();
   data->SetNumberOfComponents (3);
+  
   vtkIdType nr_points = cloud_->width * cloud_->height;
+  
+  if (!data->Resize(nr_points))
+    PCL_ERROR("Failed to allocate space for points in VTK array");
```

Comment:
Should we throw here?

Reply:
What to throw? Do we have a PCLOutOfMemoryException or something similar?

Reply:
Your solution looks good.